### PR TITLE
Add a REPL client/server

### PR DESCRIPTION
**Pull request by zardyh**
Amc REPLs will listen (on port 5478, pass port 0 to disable), and clients can connect and run code there a lot faster than spinning up a new REPL[citation needed].

### CODE REVIEW

File: compiler/Repl.hs [hunk 1]
```diff
@@ -452,19 +468,96 @@ execFile name line = do
...
+
+      handleReplLine line =
+        if | T.null line -> pure ()
+           | ':' == T.head line -> uncurry (execCommand Nothing) . span (/=' ') $ T.unpack (T.tail line)
+           | otherwise -> () <$ execString "=<remote command>" line
+
+
+  nulldev <- openFile "/dev/null" WriteMode
+  bracket getSock Net.close $ \sock -> do
+    state <- execStateT (loadFiles files) (state { outputHandle = nulldev })
```
Comment (SquidDev) [inline compiler/Repl.hs]: OK, this seems crazy dubious.

Comment (zardyh) [reply compiler/Repl.hs]: oh yeah it really is but I couldn't think of any better way. we don't want it to print the loading message twice (once for the server, once for the repl itself), so I just did that instead

Comment (zardyh) [reply compiler/Repl.hs]: ah quick fix load once before starting either I suppose

Comment (zardyh) [reply compiler/Repl.hs]: gimme a second I'm on it

**Pull request merged**